keras_nlp/src/models/retinanet/anchor_generator.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2024 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+import math
+
+import keras
+from keras import ops
+
+from keras_nlp.src.bounding_box.converters import convert_format
+
+
+class Anchor(keras.layers.Layer):
+    """A layer used for anchor-based object detectors.
+
+    Args:
+        min_level: integer number of minimum level of the output feature pyramid.
+        max_level: integer number of maximum level of the output feature pyramid.
+        num_scales: integer number representing intermediate scales added on each
+            level. For instances, num_scales=2 adds one additional intermediate
+            anchor scales [2^0, 2^0.5] on each level.
+        aspect_ratios: list of float numbers representing the aspect ratio anchors
+            added on each level. The number indicates the ratio of width to height.
+            For instances, aspect_ratios=[1.0, 2.0, 0.5] adds three anchors on each
+            scale level.
+        anchor_size: float number representing the scale of size of the base
+            anchor to the feature stride 2^level.
+    Call arguments:
+        image: an image with shape `[B, H, W, C]` or `[H, W, C]`
+        image_shape: a list of integer numbers or Tensors representing [height,
```

Comment:
Why do we need to call with `image_shape`? Isn't `ops.shape(image)` the image shape?

Reply:
Yes, the option of just passing the image_shape was present in keras-cv so included it. We can ignore it.

Reply:
Done